src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -3646,4 +3646,175 @@ public Long hstrlen(final String key, final String field) {
     return client.getIntegerReply();
   }
 
+  @Override
+  public EntryID xadd(final String key, final EntryID id, final Map<String, String> hash) {
+    return xadd(key, id, hash, Long.MAX_VALUE, true);
+  }
+  
+  @Override
+  public EntryID xadd(final String key, EntryID id, final Map<String, String> hash, final long maxLen, final boolean exactMaxLen) {
+    checkIsInMultiOrPipeline();
+    client.xadd(key, id, hash, maxLen, exactMaxLen);
+    String result = client.getBulkReply();
+    return new EntryID(result);
+  }
+
+  @Override
+  public Long xlen(final String key) {
+    checkIsInMultiOrPipeline();
+    client.xlen(key);
+    return client.getIntegerReply();
+  }
+
+  @Override
+  public List<StreamEntry> xrange(final String key, final EntryID start, final EntryID end, final int count) {
```

Comment:
make `count` optional (Integer). sometimes we'd have to allow something even if it's risky. e.g. `keys` command

Reply:
a user can always send Integer.MAX_VALUE